### PR TITLE
[WIP] Orderbook

### PR DESCRIPTION
**Pull request by femtotrader**

### CODE REVIEW

**Comment by femtotrader** [comment]
This is just a quick copy/paste of quite old code of a private project...
some cleanup is required but it can be a first start.

**Comment by roshii** [comment]
It's all a WIP for now.
Let just build on it !

File: src/orderbook/base.jl [hunk 1]
```diff
@@ -0,0 +1,41 @@
+abstract type AbstractOrderBook{Tprice, Tvolume} <: AbstractPriceData{Tprice, Tvolume} end
+
+struct OrderBookException <: Exception
+    msg::String
+end
+
+function price(ob::AbstractOrderBook, volume::Volume; raise=false)
```
Comment (femtotrader) [inline src/orderbook/base.jl:7]: Maybe I should have wrote something like

    price(ob::AbstractOrderBook{Tprice, Tvolume}, volume::Tvolume; raise=false) where {Tprice, Tvolume}

as I said it's a very old code and I wasn't very fluent using parametric types

File: src/orderbook/base.jl [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    if volume == Volume(0)
```
Comment (femtotrader) [inline src/orderbook/base.jl:8]: should probably be

    if volume == Tvolume(0)

File: src/orderbook/base.jl [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        midpoint = (bid(ob)[1] + ask(ob)[1]) / 2
+        midpoint, volume
+    else
+        if volume > Volume(0)
```
Comment (femtotrader) [inline src/orderbook/base.jl:12]: same here

File: src/orderbook/base.jl (hunk 1 above)
Comment (roshii) [reply src/orderbook/base.jl:7]: https://github.com/JuliaFinance/MarketObservables.jl/issues/5#issuecomment-486108913

File: src/orderbook/base.jl (hunk 2 above)
Comment (roshii) [reply src/orderbook/base.jl:8]: https://github.com/JuliaFinance/MarketObservables.jl/issues/5#issuecomment-486108913

**Pull request merged**